commandapi-platforms/commandapi-velocity/commandapi-velocity-plugin/src/main/java/dev/jorel/commandapi/config/VelocityConfigurationAdapter.java

```diff
@@ -79,58 +78,28 @@ public CommentedConfigurationNode config() {
+	public ConfigurationAdapter<ConfigurationNode, DefaultVelocityConfig> complete() {
 		return this;
 	}
 
 	@Override
-	public ConfigurationAdapter<ConfigurationNode> createNew() {
-		return new VelocityConfigurationAdapter(loader, loader.createNode(), DefaultVelocityConfig.createDefault());
+	public ConfigurationAdapter<ConfigurationNode, DefaultVelocityConfig> createNew() {
+		return new VelocityConfigurationAdapter(loader, loader.createNode(), createDefaultConfig());
 	}
 
 	@Override
-	public void saveDefaultConfig(File directory, File configFile, Logger logger) {
-		YamlConfigurationLoader configLoader = YamlConfigurationLoader.builder()
-			.nodeStyle(NodeStyle.BLOCK)
-			.file(configFile)
-			.build();
-		DefaultVelocityConfig defaultConfig = DefaultVelocityConfig.createDefault();
-		ConfigGenerator configGenerator = ConfigGenerator.createNew(defaultConfig);
-		if (!directory.exists()) {
-			boolean createdDirectory = directory.mkdirs();
-			if (!createdDirectory) {
-				logger.severe("Failed to create directory for the CommandAPI's config.yml file!");
-			}
-			try {
-				ConfigurationAdapter<ConfigurationNode> velocityConfigurationAdapter = new VelocityConfigurationAdapter(configLoader, configLoader.createNode(), defaultConfig);
-				configGenerator.populateDefaultConfig(velocityConfigurationAdapter);
-				configLoader.save(velocityConfigurationAdapter.config());
-			} catch (IOException e) {
-				logger.severe("Could not create default config file! This is (probably) a bug.");
-				logger.severe("Error message: " + e.getMessage());
-				logger.severe("Stacktrace:");
-				for (StackTraceElement element : e.getStackTrace()) {
-					logger.severe(element.toString());
-				}
-			}
-		} else {
-			try {
-				// If the config does exist, update it if necessary
-				CommentedConfigurationNode existingYamlConfig = configLoader.load();
-				ConfigurationAdapter<ConfigurationNode> existingConfig = new VelocityConfigurationAdapter(configLoader, existingYamlConfig, defaultConfig);
-				ConfigurationAdapter<ConfigurationNode> updatedConfig = configGenerator.generateWithNewValues(existingConfig);
-				if (updatedConfig != null) {
-					configLoader.save(updatedConfig.config());
-				}
-			} catch (IOException e) {
-				logger.severe("Could not update config! This is (probably) a bug.");
-				logger.severe("Error message: " + e.getMessage());
-				logger.severe("Stacktrace:");
-				for (StackTraceElement element : e.getStackTrace()) {
-					logger.severe(element.toString());
-				}
-			}
-		}
+	public DefaultVelocityConfig createDefaultConfig() {
+		return DefaultVelocityConfig.createDefault();
+	}
+
+	@Override
+	public ConfigurationAdapter<ConfigurationNode, DefaultVelocityConfig> loadFromFile(File file) throws IOException {
+		return new VelocityConfigurationAdapter(loader, loader.load(), createDefaultConfig());
+	}
+
+	@Override
+	public void saveToFile(File file) throws IOException {
+		loader.save(config);
 	}
```

Comment:
Yeah, ideally I wouldn't have to add the file as a parameter to those methods, however Bukkit works that way so the file parameter is only there for Bukkit since Velocity can just use the loader's file. Really wanted to to avoid to create a loader twice which was criticism of yours before.

Reply:
> Another alternative: instead of passing the `File directory` and `File configFile` to `saveDefaultConfig`, those could perhaps be set when constructing `BukkitConfigurationAdapter` and `VelocityConfigurationAdapter`. So neither `loadFromFile` nor `saveToFile` would need a `File` parameter. `VelocityConfigurationAdapter` is already doing this, so this would be the option of updating `BukkitConfigurationAdapter` to match the implementations here.

So maybe this option then? Like currently when `VelocityConfigurationAdapter` is constructed, it is given a `YamlConfigurationLoader` which represents the `File` being edited. On Bukkit it could do a similar thing where I think it would just store the `File` directly as a field and use that in `loadFromFile` and `saveToFile`.


> Really wanted to to avoid to create a loader twice which was criticism of yours before.

Yea, I guess I in this comment https://github.com/JorelAli/CommandAPI/pull/596#discussion_r1791795600 I was more so looking at how `VelocityConfigurationAdapter` and `CommandAPIMain` were using identical code to build their `YamlConfigurationLoader`. If we updated the settings in one on those places, we'd have to remember to update it in the other place. I wouldn't expect creating a `YamlConfigurationLoader` is an especially expensive task that should be avoided, as long as we are sure they are being created the same way each time, which is what I was thinking with the static `createLoader` method.

But yeah, I think there are many possible ways to make the method behavior here consistent between the platforms.